opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -398,7 +397,55 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
-class Span(trace_api.Span):
+class ReadableSpan:
+    """Provides read-only access to span attributes"""
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def context(self) -> trace_api.SpanContext:
+        return self._context
+
+    @property
+    def kind(self) -> trace_api.SpanKind:
+        return self._kind
+
+    @property
+    def parent(self) -> Optional[trace_api.SpanContext]:
+        return self._parent
+
+    @property
+    def start_time(self) -> Optional[int]:
+        return self._start_time
+
+    @property
+    def end_time(self) -> Optional[int]:
+        return self._end_time
+
+    @property
+    def status(self) -> trace_api.Status:
+        return self._status
+
+    @property
+    def attributes(self) -> types.Attributes:
+        return self._attributes
```

Comment:
I believe these attributes will be mutable, as it's returning back the raw dict. Is that ok?

Same with events and links.

Reply:
That's correct, I wasn't sure if this would be an issue or not.

Reply:
We should use a mappingproxytype to make sure these are immutable as well.